### PR TITLE
T15065 (part 1)

### PR DESCRIPTION
**Pull request by tchx84**
https://phabricator.endlessm.com/T15065

### CODE REVIEW

File: ekncontent/ekncontent/eknc-domain.c [hunk 1]
```diff
@@ -662,6 +662,10 @@ on_object_response (GObject *source,
   QueryState *state = g_task_get_task_data (task);
   GError *error = NULL;
 
+  // do nothing if we already returned error
+  if (g_task_had_error(task))
```
Comment (mattdangerw) [inline ekncontent/ekncontent/eknc-domain.c]: Could this more generally just be `g_task_get_completed`?

Comment (mattdangerw) [inline ekncontent/ekncontent/eknc-domain.c]: That way if we hit an early return that's not an error we are still covered.

Comment (tchx84) [inline ekncontent/ekncontent/eknc-domain.c]: sure, will change it now!

**Comment by mattdangerw** [comment]
CR pass! (not merging yet though)

**Comment by mattdangerw** [comment]
Heh nice catch. Looks good! CR pass again

**Pull request merged**